src/lib/y2partitioner/widgets/delete_disk_partition_button.rb

```diff
@@ -28,18 +28,22 @@ def label
       def handle
         device = @device || id_to_device(@table.value)
 
-        # TODO: check for children that will die and if there are, use confirm_recursive_delete
-        ret = Yast::Popup.YesNo(
-          # TRANSLATORS %s is device to be deleted
-          format(_("Really delete %s?"), device.name)
-        )
+        if device.nil?
+          Yast::Popup.Error(_("No device selected"))
+          return nil
+        end
 
-        return nil unless ret
+        return nil unless confirm(device)
 
         partition_table = device.partition_table
         if device.is?(:disk)
```

Comment:
yes, so far no support for s390 in partitioner, also we should support here other devices like LVM, or RAD which can be deleted.